order-history-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/orderhistoryservice/domain/CustomerViewRepository.java

```diff
@@ -1,41 +1,85 @@
 package io.eventuate.examples.tram.ordersandcustomers.orderhistoryservice.domain;
 
+import com.mongodb.BasicDBObject;
+import com.mongodb.client.MongoClient;
 import io.eventuate.examples.tram.ordersandcustomers.commondomain.Money;
 import io.eventuate.examples.tram.ordersandcustomers.commondomain.OrderState;
 import io.eventuate.examples.tram.ordersandcustomers.orderhistory.common.CustomerView;
 import io.eventuate.examples.tram.ordersandcustomers.orderhistory.common.OrderInfo;
-import org.springframework.data.mongodb.core.MongoTemplate;
-import org.springframework.data.mongodb.core.query.Query;
-import org.springframework.data.mongodb.core.query.Update;
-
+import org.bson.Document;
 import java.util.Optional;
 
-import static org.springframework.data.mongodb.core.query.Criteria.where;
-
-public class CustomerViewRepository {
+public class CustomerViewRepository extends AbstractRepository {
 
-  private MongoTemplate mongoTemplate;
-
-  public CustomerViewRepository(MongoTemplate mongoTemplate) {
-    this.mongoTemplate = mongoTemplate;
+  public CustomerViewRepository(MongoClient mongoClient) {
+    super(mongoClient, "customers");
   }
 
   public Optional<CustomerView> findById(Long customerId) {
-    return Optional.ofNullable(mongoTemplate.findById(customerId, CustomerView.class));
+    Document customerDocument = collection()
```

Comment:
This pattern is repeated

```
    Document customerDocument = collection()
            .find(new BasicDBObject("_id", ID))
            .first();

    return Optional
            .ofNullable(customerDocument)

```

Make use of `AbstractRepository` and define a `findOne(ID)` method, which returns `Optional<Document>`

Reply:
ok